proto/cosmos/base/abci/v1beta1/abci.proto

```diff
@@ -175,3 +179,35 @@ message SearchBlocksResult {
   // List of blocks in current page
   repeated tendermint.types.Block blocks = 6;
 }
+
+// LegacyRequestBeginBlock defines a wrapper around the now deprecated
+// RequestBeginBlock type.
+message LegacyRequestBeginBlock {
+  // TODO: Add legacy fields.
+  //
+  // Ref: https://github.com/cosmos/cosmos-sdk/issues/12272
+}
+
+// LegacyResponseBeginBlock defines a wrapper around the now deprecated
+// ResponseBeginBlock type.
+message LegacyResponseBeginBlock {
+  // TODO: Add legacy fields.
+  //
+  // Ref: https://github.com/cosmos/cosmos-sdk/issues/12272
+}
+
+// LegacyRequestEndBlock defines a wrapper around the now deprecated
+// RequestEndBlock type.
+message LegacyRequestEndBlock {
+  // TODO: Add legacy fields.
+  //
+  // Ref: https://github.com/cosmos/cosmos-sdk/issues/12272
+}
+
+// LegacyResponseEndBlock defines a wrapper around the now deprecated
+// ResponseEndBlock type.
+message LegacyResponseEndBlock {
+  // TODO: Add legacy fields.
+  //
+  // Ref: https://github.com/cosmos/cosmos-sdk/issues/12272
+}
```

Comment:
i dont think we need these types. modules can opt to use core api https://github.com/cosmos/cosmos-sdk/blob/9b0278408ce4be82a251beae280e2c0c9c9fab62/core/appmodule/module.go#L48 which doesnt need these types anymore. the only one that we may need is endblock because we dont have a return valiadator set updates yet.

Reply:
For `BeginBlock`, yes, we can use `Context`. However, `EndBlock` needs to return values to CometBFT, namely, validator updates and consensus params. Feel free to define a type that contains these.